cmake/Modules/FindODBC.cmake

```diff
@@ -0,0 +1,215 @@
+#
+# Consults the following vars (if set):
+#     ODBC_PROVIDER - one of UnixODBC, iODBC, or MDAC
+#
+#     ODBC_DIR
+#     ODBC_CONFIG_SCRIPT
+#
+#     ODBC_SKIP_CONFIG_SCRIPT
+#     ODBC_SKIP_PKG_CONFIG
+#     ODBC_SKIP_BREW
+#
+#     ODBC_<PROVIDER>_DIR
+#     ODBC_<PROVIDER>_CONFIG_SCRIPT
+#
+#     ODBC_<PROVIDER>_SKIP_CONFIG_SCRIPT
+#     ODBC_<PROVIDER>_SKIP_PKG_CONFIG
+#     ODBC_<PROVIDER>_SKIP_BREW
+#
+# Defines the following targets:
+#     ODBC::App
+#     ODBC::Driver
+#
+# Defines (some of) the following vars:
+#     ODBC_FOUND
+#     ODBC_PROVIDER
+#
+#     ODBC_DIR
+#     ODBC_CONFIG_SCRIPT
+#
+#     ODBC_<PROVIDER>_FOUND
+#     ODBC_<PROVIDER>
+#
+#     ODBC_<PROVIDER>_DIR
+#     ODBC_<PROVIDER>_CONFIG_SCRIPT
+#
+#     ODBC_UNIXODBC_ISQL
+#     ODBC_UNIXODBC_IUSQL
+#
+#     ODBC_IODBC_IODBCTEST
+#     ODBC_IODBC_IODBCTESTW
+#
+#     ODBC_APP_DEFINES
+#     ODBC_APP_INCLUDE_DIRS
+#     ODBC_APP_COMPILER_FLAGS
+#     ODBC_APP_LINKER_FLAGS
+#
+#     ODBC_DRIVER_DEFINES
+#     ODBC_DRIVER_INCLUDE_DIRS
+#     ODBC_DRIVER_COMPILER_FLAGS
+#     ODBC_DRIVER_LINKER_FLAGS
+#
+#     ODBC_<PROVIDER>_APP_DEFINES
+#     ODBC_<PROVIDER>_APP_INCLUDE_DIRS
+#     ODBC_<PROVIDER>_APP_COMPILER_FLAGS
+#     ODBC_<PROVIDER>_APP_LINKER_FLAGS
+#
+#     ODBC_<PROVIDER>_DRIVER_DEFINES
+#     ODBC_<PROVIDER>_DRIVER_INCLUDE_DIRS
+#     ODBC_<PROVIDER>_DRIVER_COMPILER_FLAGS
+#     ODBC_<PROVIDER>_DRIVER_LINKER_FLAGS
+#
+
+if (WIN32)
+    set (_providers_to_try MDAC;)
+elseif (APPLE)
+    set (_providers_to_try iODBC;UnixODBC)
+else ()
+    set (_providers_to_try UnixODBC;iODBC)
+endif ()
+
+if (ODBC_PROVIDER)
+    string (TOUPPER "${ODBC_PROVIDER}" _provider_uc)
+    if ("${_provider_uc}" STREQUAL "MDAC")
+        set (_providers_to_try MDAC;)
+    elseif ("${_provider_uc}" STREQUAL "UNIXODBC")
+        set (_providers_to_try UnixODBC;)
+    elseif ("${_provider_uc}" STREQUAL "IODBC")
+        set (_providers_to_try iODBC;)
+    else ()
+        message (FATAL_ERROR "Unknown ODBC provider: ${ODBC_PROVIDER}")
+    endif ()
+endif ()
```

Comment:
IMO, it make a bit more sense to re-arrange this piece as follows:

```
if (ODBC_PROVIDER)
    string (TOUPPER "${ODBC_PROVIDER}" _provider_uc)
    if ("${_provider_uc}" STREQUAL "MDAC")
        set (_providers_to_try MDAC;)
    elseif ("${_provider_uc}" STREQUAL "UNIXODBC")
        set (_providers_to_try UnixODBC;)
    elseif ("${_provider_uc}" STREQUAL "IODBC")
        set (_providers_to_try iODBC;)
    else ()
        message (FATAL_ERROR "Unknown ODBC provider: ${ODBC_PROVIDER}")
    endif ()
# Platform-default provides
elseif (WIN32)
    set (_providers_to_try MDAC;)
elseif (APPLE)
    set (_providers_to_try iODBC;UnixODBC)
else ()
    set (_providers_to_try UnixODBC;iODBC)
endif ()
```

Reply:
Done.